### PR TITLE
Add Instructions view

### PR DESCRIPTION
**Pull request by abitblue**
In the event that we detect a randomized MAC address, we need to instruct the user to disable on both the Joining network and the Access network.

TODO (still):
- Add mechanism to detect randomized MAC address

### CODE REVIEW

File: login/static/style.css [hunk 1]
```diff
@@ -0,0 +1,230 @@
+html {
+    --color-techgold: #C4B55F;
+    box-sizing: border-box;
+}
+
+body {
+    font-family: sans-serif;
+    margin: 1%;
+}
```
Comment (harveyj123) [inline login/static/style.css:9]: you can combine these 2  like html, body {   }

File: login/static/style.css [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+.centerDiv {
+    text-align: center;
+    margin: auto;
+    width: 85%;
+}
+
+img {
+    height: 1.5em;
+    vertical-align: middle;
+}
```
Comment (harveyj123) [inline login/static/style.css:20]: use classes to for img if your going to do it for mutiple imgs this css sucks no cap. use .img bozo. Use Ids for specipic sizing imgs also use  percages for imgs

File: login/static/style.css [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+
+.button2{
+    margin-bottom: 4vh !important;
+}
+
+.bodyTextInsignificant {
+    color: gray;
+    font-size: 1.5vh !important;
+}
```
Comment (harveyj123) [inline login/static/style.css:65]: !important https://stackoverflow.com/questions/3706819/what-are-the-implications-of-using-important-in-css

File: login/static/style.css [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+    font-size: 4.5vh;
+    font-weight: bold;
+    margin: 1vh 1vh 2.25vh 1vh;
+    display: inline-block;
+}
+
+@media only screen and (max-width: 480px) {
+    body {
+        margin: 3% 1% 1%;
+    }
```
Comment (harveyj123) [inline login/static/style.css:40]: WTF IS THIS ?

File: login/static/style.css [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+    vertical-align: middle;
+}
+
+.headerText {
+    font-size: 4.5vh;
+    font-weight: bold;
+    margin: 1vh 1vh 2.25vh 1vh;
+    display: inline-block;
+}
+
```
Comment (harveyj123) [inline login/static/style.css:36]: USE flexbox TYSM MY EYES HURT

File: login/static/style.css [hunk 6]
```diff
@@ -0,0 +1,230 @@
+html {
```
Comment (harveyj123) [inline login/static/style.css:1]: revert and use flexbox tysm!

File: templates/login/base.html [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+            <img src="{% static 'logo1.svg' %}" srcset="{% static 'logo1.svgz' %}" alt="SITHS Logo">
+            <img src="{% static 'logo2.svg' %}" srcset="{% static 'logo2.svgz' %}" alt="SITHS Technology Dept">
+            <span class="headerText">Device Registration</span>
+        </div>
+    {% endblock %}
+
+    {% block content %} {% endblock %}
+
+    {% block footer %}
+        <div class="footerStroke"></div>
```
Comment (harveyj123) [inline templates/login/base.html:25]: cool screw screen readers

File: templates/login/instructions/fotter_buttons.html [hunk 8]
```diff
@@ -0,0 +1,3 @@
+<br> <br>
+<a href="{% url 'instructions' %}"><button class="button"><span class="buttonSpan">Go Back</span></button></a>
+<a href="{% url 'index' %}"><button class="button"><span class="buttonSpan">Continue</span></button></a>
```
Comment (harveyj123) [inline templates/login/instructions/fotter_buttons.html:3]: using br lol

File: templates/login/instructions/with_os.html [hunk 9]
```diff
@@ -0,0 +1,25 @@
+{% extends "login/base.html" %}
+
+{% block title %}Instructions{% endblock %}
+
+{% block content %}
+    <header class="headerText">Disabling MAC Address Randomization on {{ device_os_verbose }}</header>
+    <br>
```
Comment (harveyj123) [inline templates/login/instructions/with_os.html:7]: using br

File: templates/login/instructions/without_os.html [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+
+    <a href="{% url 'instructions' %}?os=windows"><button class="button"><span class="buttonSpan">Windows</span></button></a>
+    <br>
+    <a href="{% url 'instructions' %}?os=mac"><button class="button"><span class="buttonSpan">Macintosh</span></button></a>
+    <br>
+    <a href="{% url 'instructions' %}?os=ios"><button class="button"><span class="buttonSpan">iOS / iPadOS</span></button></a>
+    <br>
+    <a href="{% url 'instructions' %}?os=android"><button class="button2 button"><span class="buttonSpan">Android</span></button></a>
+
+    <br>
```
Comment (harveyj123) [inline templates/login/instructions/without_os.html:21]: HAHAHAHHAH ?

File: templates/login/instructions/os_android.html [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+    {% else %} {# version_check == "right" #}
+        <p class="bodyText"><b>Step 1:</b> Navigate to the Wi-Fi settings of your Android Device.</p>
+        <img class="instructions-img" src="{% static 'instructions/wifi_network_android.jpg' %}" alt="wifi and network settings on android">
+        <p class="bodyText"><b>Step 2:</b> Click on the Gear next to the Wifi name.</p>
+        <img class="instructions-img" src="{% static 'instructions/gear_android.jpg' %}" alt="gear icon next to the wifi network">
+        <p class="bodyText"><b>Step 3:</b> Open up Advanced settings in the network.</p>
+        <img class="instructions-img" src="{% static 'instructions/details_android.jpg' %}" alt="advanced settings dropdown">
+        <p class="bodyText"><b>Step 4:</b> Locate the Privacy setting and make sure you are using your Device MAC.</p>
+        <img class="instructions-img" style="margin-bottom: 5vh" src="{% static 'instructions/advanced_android.jpg' %}" alt="location of the privacy setting">
+        <br>
```
Comment (harveyj123) [inline templates/login/instructions/os_android.html:22]: killing spree

File: login/views/homepage.py [hunk 12]
```diff
@@ -8,3 +10,28 @@ class Index(View):
...
+
+        return render(request, 'login/instructions/with_os.html', {
+            'device_os': device_os,
+            'version_check': version_check,
+            'device_os_verbose': {
+                'windows': 'Windows',
+                'mac': 'macOS',
+                'android': 'Android',
+                'ios': 'iOS/iPadOS',
+            }.get(device_os, 'Unknown')
```
Comment (harveyj123) [inline login/views/homepage.py:36]: need to write test for the class Instructions

File: login/views/homepage.py [hunk 13]
```diff
@@ -8,3 +10,28 @@ class Index(View):
     def get(self, request: HttpRequest):
         # TODO: fill!
```
Comment (harveyj123) [inline login/views/homepage.py:11]: tests

File: login/views/homepage.py [hunk 14]
```diff
@@ -8,3 +10,28 @@ class Index(View):
...
+class Instructions(View):
+    """If the user is using randomized MAC addresses, these instructions will direct them to disable it"""
+    template_name = 'instructions/with_os.html'
+
+    def get(self, request: HttpRequest):
+        device_os = request.GET.get('os', None)
+        version_check = request.GET.get('version_check', None)
+
+        # If device_os is not a GET parameter, ask user which OS is in use.
+        # OR, if
```
Comment (harveyj123) [inline login/views/homepage.py:24]: inc commet

**Review by harveyj123 — CHANGES_REQUESTED** [review]
overall we sould be using samatic html and css with flexbox etc also we sould write tests for all functions/ classes

File: login/static/style.css (hunk 1 above)
Comment (IvanZ505) [reply login/static/style.css:9]: sure! you're so smart and amazing, Harvey!

File: login/static/style.css (hunk 2 above)
Comment (IvanZ505) [reply login/static/style.css:20]: 👍

File: login/static/style.css (hunk 3 above)
Comment (IvanZ505) [reply login/static/style.css:65]: 👍

File: login/views/homepage.py (hunk 13 above)
Comment (IvanZ505) [reply login/views/homepage.py:11]: theres nothing to test bro

**Review by IvanZ505 — APPROVED** [review]
Push through for update to branch.
Will make issue for the CSS changes and tests.

**Review by harveyj123 — APPROVED** [review]

**Pull request merged**